cloudbuild-test.yaml

```diff
@@ -0,0 +1,94 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+# set the project ID in the k8s manifest for service account/workload identity
+- name: 'ubuntu'
+  id: Replace
+  script: |
+    #!/usr/bin/env bash
+    sed -i "s/%GCLOUD_PROJECT%/${PROJECT_ID}/g" collector/*
+  env:
+  - 'PROJECT_ID=$PROJECT_ID'
+
+# create a GKE cluster
+- name: 'gcr.io/cloud-builders/gcloud'
+  id: CreateCluster
+  args:
+  - 'container'
+  - 'clusters'
+  - 'create'
+  - '${COMMIT_SHA}'
+  - '--zone=${LOCATION}'
+
+# deploy the manifests in GKE
+- name: 'gcr.io/cloud-builders/kubectl'
+  id: Deploy
+  args:
+  - 'apply'
+  - '-f'
+  - './collector/.'
+  env:
+  - 'CLOUDSDK_COMPUTE_REGION=${LOCATION}'
+  - 'CLOUDSDK_CONTAINER_CLUSTER=${COMMIT_SHA}'
+
+# assert that all pods are up and running
+- name: 'gcr.io/cloud-builders/kubectl'
+  id: AssertReady
+  args:
+  - 'wait'
+  - '--for=condition=Ready'
+  - '--timeout=300s'
+  - '-n'
+  - 'opentelemetry'
+  - 'pods'
+  - '--all'
+  env:
+  - 'CLOUDSDK_COMPUTE_REGION=${LOCATION}'
+  - 'CLOUDSDK_CONTAINER_CLUSTER=${COMMIT_SHA}'
+
+# delete everything in the opentelemetry namespace
+- name: 'gcr.io/cloud-builders/kubectl'
+  id: TearDown
```

Comment:
we should set `allowFailure: true` on `TearDown` and `DeleteNamespace` so that the rest of the cleanup steps will continue even if these fail for some reason (https://cloud.google.com/build/docs/release-notes#April_26_2023)

Alternatively, we could get rid of `TearDown` and `DeleteNamespace` and just let `DeleteCluster` do all the cleanup. That might reduce flakes and save time in each test

Reply:
Went with the second suggestion and now we just have the `DeleteCluster` step to tear it all down